extensions/wikia/Discussions/maintenance/ForumDumper.php

```diff
@@ -0,0 +1,246 @@
+
+	const COLUMNS_PAGE = [
+		"page_id",
+		"namespace",
+		"raw_title",
+		"is_redirect",
+		"is_new",
+		"touched",
+		"latest_revision_id",
+		"length",
+		"parent_page_id",
+		"parent_comment_id",
+		"last_child_comment_id",
+		"archived_ind",
+		"deleted_ind",
+		"removed_ind",
+		"locked_ind",
+		"protected_ind",
+		"sticky_ind",
+		"first_revision_id",
+		"last_revision_id",
+		"comment_timestamp"
+	];
+
+	const COLUMNS_REVISION = [
+		"revision_id",
+		"page_id",
+		"page_namespace",
+		"title",
+		"user_type",
+		"user_identifier",
+		"timestamp",
+		"is_minor_edit",
+		"is_deleted",
+		"length",
+		"parent_id",
+		"text_flags",
+		"comment",
+		"raw_content",
+		"content"
+	];
+
+	const COLUMNS_VOTE = [
+		"page_id",
+		"user_identifier",
+		"timestamp",
+	];
+
+	private $pages = [];
+	private $revisions = [];
+	private $votes = [];
+
+	public function addPage( $id, $data ) {
+		$this->pages[$id] = $data;
+	}
+
+	public function addRevision( $data ) {
+		$this->revisions[] = $data;
+	}
+
+	public function addVote( $data ) {
+		$this->votes[] = $data;
+	}
+
+	public function getPages() {
+		if ( !empty( $this->pages ) ) {
+			return $this->pages;
+		}
+
+		$dumper = $this;
+		$dbh = wfGetDB( DB_SLAVE );
+		( new \WikiaSQL() )
+			->SELECT_ALL()
+			->FROM( self::TABLE_PAGE )
+			->LEFT_JOIN( self::TABLE_COMMENTS )->ON( 'page_id', 'comment_id' )
+			->WHERE( 'page_namespace' )->IN( self::FORUM_NAMESPACES )
+			->runLoop( $dbh, function ( &$pages, $row ) use ( $dumper ) {
+				$dumper->addPage( $row->page_id, [
+						"page_id" => $row->page_id,
+						"namespace" => $row->page_namespace,
+						"raw_title" => $row->page_title,
+						"is_redirect" => $row->page_is_redirect,
+						"is_new" => $row->page_is_new,
+						"touched" => $row->page_touched,
+						"latest_revision_id" => $row->page_latest,
+						"length" => $row->page_len,
+						"parent_page_id" => $row->parent_page_id,
+						"parent_comment_id" => $row->parent_comment_id,
+						"last_child_comment_id" => $row->last_child_comment_id,
+						"archived_ind" => $row->archived ?: 0,
+						"deleted_ind" => $row->deleted ?: 0,
+						"removed_ind" => $row->removed ?: 0,
+						"locked_ind" => $row->locked ?: 0,
+						"protected_ind" => $row->protected ?: 0,
+						"sticky_ind" => $row->sticky ?: 0,
+						"first_revision_id" => $row->first_rev_id,
+						"last_revision_id" => $row->last_rev_id,
+						"comment_timestamp" => $row->last_touched
+					] );
+			} );
+
+		return $this->pages;
+	}
+
+	public function getRevisions() {
+		if ( !empty( $this->revisions ) ) {
+			return $this->revisions;
+		}
+		
+		$pageIds = array_keys( $this->getPages() );
+
+		$dumper = $this;
+		$dbh = wfGetDB( DB_SLAVE );
+		( new \WikiaSQL() )
+			->SELECT_ALL()
+			->FROM( self::TABLE_REVISION )
+			->JOIN( self::TABLE_TEXT )->ON( 'rev_text_id', 'old_id' )
+			->WHERE( 'rev_page' )->IN( $pageIds )
+			->runLoop( $dbh, function ( &$revisions, $row ) use ( $dumper ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "$revisions". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)